### DISCUSSION TITLE
Filtering based on a dynamic nullable value

### DISCUSSION THREAD

**Discussion by TimurKr** [supabase/supabase#16204] [Questions]
I need to fetch data from my database and filter them based on a certain nullable bigint column. I would normaly use the `eq.<value>` filer, but that doesnt work when `value` is null, that requires `is.null`. How can I filter based on the `value`, when the `value` is dynamic and can be null during runtime?

**Comment by GaryAustin1** [comment]
At a minimum you can use conditional chaining...
![image](https://github.com/supabase/supabase/assets/54564956/7ddc61d2-6037-49f3-b36e-04d886760ef6)
From javascript client docs.

**Reply by TimurKr** [reply]
Thanks, that would work but seems like a lot of code for such simple scenario. I am working with flutter and in that library (probably very similar in javascript) it seems like a simple 2 line code change to add support for null values with the eq filter. Where would be the right place to write such request? Or is there a way for me to do this? I have never contributed to an open-source code before, so I am not sure how.

**Reply by GaryAustin1** [reply]
Well there are two possible points the client side or the API side.
supabase/postgrest-js and supabase/supabase-flutter (I don't know the package structure under flutter for postgrest) could detect null and then change the call to PostgREST (the API).
Or [PostgREST](https://github.com/PostgREST/postgrest) is the actual API Supabase uses.
At a minimum you could generate an issue if it does not exist.  As far as changes they do take PR's but Supabase tries to keep clients similar and follow supabase-js from what I have seen.  So might have to change in several clients.

**Reply by GaryAustin1** [reply]
Also note that you have to convert .eq to .is somewhere because Postgres the database does not work with `= null` it requires `is null` at the SQL level.

**Reply by TimurKr** [reply]
Great, thanks for the reply. I created a [pull request](https://github.com/supabase/supabase-flutter/pull/580) for the flutter library. I'll see if that gets accepted. I am not sure if that was the best place to implement this, but seems like the easiest. Might add it to the js as well if I get a response.
